### DISCUSSION TITLE
PThread Priority Inheritance

### DISCUSSION THREAD

**Discussion by SterlingPeet** [nasa/fprime#999] [Q&A] [answered]
What is the significance of the PThread priority inheritance feature in F Prime?  Can I just remove a few lines of code and reasonably expect that things will work ok in a lightly loaded system?

For reference, I am looking into the problem that was mentioned on the Google Groups a while ago: 

https://groups.google.com/g/fprime-community/c/ynOrgwfApAE

Thanks,

Sterling Peet

**Comment by timcanham** [comment, marked as answer]
Priority inheritance solves a particular deadlock problem in RTOSes, priority inversion:

https://en.wikipedia.org/wiki/Priority_inversion

If a low-priority task is holding a mutex in a priority inversion condition, it will temporarily raise the priority of of the lower priority task to free up the higher priority task. It will only do this if the mutex has that attribute enabled, which is why we always enable it in our code. In "regular" Linux userspace, the tasks are round-robined so one task can't hold the resource forever.

Are you thinking of removing the setting to solve a particular problem?

**Reply by SterlingPeet** [reply]
Well, my target platform is an embedded Linux platform, and that feature is not supported the default PThreads library implementation.  I don't think that removing this feature is a good plan for general usage, but I also didn't want to try and implement the feature myself for this project.

Mostly I just wanted to make sure that I wasn't going break some fundamental assumption on a platform that is not running critical timing code.  It seems that should be fine for what I am doing.
